Sidecar requires RestTemplate bean

For `Finchley.BUILD-SNAPSHOT` Sidecar requires a bean of type `RestTemplate` to be explicitly set

```
java.lang.IllegalStateException: Failed to load ApplicationContext
Caused by: org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'healthCheckHandler' defined in org.springframework.cloud.netflix.sidecar.SidecarConfiguration$EurekaInstanceConfigBeanConfiguration: Unsatisfied dependency expressed through method 'healthCheckHandler' parameter 0; nested exception is org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'localApplicationHealthIndicator': Unsatisfied dependency expressed through field 'restTemplate'; nested exception is org.springframework.beans.factory.NoSuchBeanDefinitionException: No qualifying bean of type 'org.springframework.web.client.RestTemplate' available: expected at least 1 bean which qualifies as autowire candidate. Dependency annotations: {@org.springframework.beans.factory.annotation.Autowired(required=true)}
Caused by: org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'localApplicationHealthIndicator': Unsatisfied dependency expressed through field 'restTemplate'; nested exception is org.springframework.beans.factory.NoSuchBeanDefinitionException: No qualifying bean of type 'org.springframework.web.client.RestTemplate' available: expected at least 1 bean which qualifies as autowire candidate. Dependency annotations: {@org.springframework.beans.factory.annotation.Autowired(required=true)}
Caused by: org.springframework.beans.factory.NoSuchBeanDefinitionException: No qualifying bean of type 'org.springframework.web.client.RestTemplate' available: expected at least 1 bean which qualifies as autowire candidate. Dependency annotations: {@org.springframework.beans.factory.annotation.Autowired(required=true)}
```

You can check https://github.com/spring-cloud/spring-cloud-core-tests `Finchley` branch.

The failing tests are

```
demo.SidecarApplicationTests.instanceConfigHasCorrectPort
demo.SidecarApplicationTests.contextLoads
```

--- a/spring-cloud-netflix-sidecar/src/main/java/org/springframework/cloud/netflix/sidecar/SidecarConfiguration.java
+++ b/spring-cloud-netflix-sidecar/src/main/java/org/springframework/cloud/netflix/sidecar/SidecarConfiguration.java
@@ -29,6 +29,7 @@ import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
 import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
 import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
 import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.boot.web.client.RestTemplateBuilder;
 import org.springframework.cloud.client.actuator.HasFeatures;
 import org.springframework.cloud.commons.util.InetUtils;
 import org.springframework.cloud.netflix.eureka.EurekaInstanceConfigBean;
@@ -166,20 +167,22 @@ public class SidecarConfiguration {
 	@Bean
 	@ConditionalOnMissingClass("org.apache.http.client.HttpClient")
 	public RestTemplate restTemplate() {
-		return new RestTemplate();
+		return new RestTemplateBuilder().build();
 	}
 
 	@Bean
 	@ConditionalOnClass(HttpClient.class)
-	@ConditionalOnProperty(value = "sidecar.accept-all-ssl-certificates")
-	public RestTemplate sslRestTemplate() {
-		CloseableHttpClient httpClient = HttpClients.custom()
-				.setSSLHostnameVerifier(new NoopHostnameVerifier())
-				.build();
-		HttpComponentsClientHttpRequestFactory requestFactory = new HttpComponentsClientHttpRequestFactory();
-		requestFactory.setHttpClient(httpClient);
-
-		return new RestTemplate(requestFactory);
+	public RestTemplate sslRestTemplate(SidecarProperties properties) {
+		RestTemplateBuilder builder = new RestTemplateBuilder();
+		if(properties.acceptAllSslCertificates()) {
+			CloseableHttpClient httpClient = HttpClients.custom()
+					.setSSLHostnameVerifier(new NoopHostnameVerifier())
+					.build();
+			HttpComponentsClientHttpRequestFactory requestFactory = new HttpComponentsClientHttpRequestFactory();
+			requestFactory.setHttpClient(httpClient);
+			builder = builder.requestFactory(() -> requestFactory);
+		}
+		return builder.build();
 	}
 
 	@Bean

